[yugan01] iP

yugan01 opened the pull request:
# Welcome to House Atreides

> I must not fear. Fear is the mind-killer. Fear is the little-death that brings total obliteration. I will face my fear. I will permit it to pass over me and through me. And when it has gone past I will turn the inner eye to see its path. Where the fear has gone there will be nothing. Only I will remain. ~ Dune

You can talk to a representative of House Atreides and give commands. Below are the features of the Atreides chatbot.

## Features
- Add Tasks
1. Todo
2. Event
3. Deadline
- Delete Tasks
- Mark Tasks as done/undone
- Find Tasks

## Main code
The `main` code is run by 
```java
public static void main(String[] args) {
    new Atreides("src/main/atreides.ui.Atreides.txt").run();
}
```
## Progress
- [x] Managing Tasks 
- [x] Managing Deadlines
- [ ]  GUI (coming soon) 🤓 

##
Try it yourself!
- **Download** the [jar file](https://github.com/yugan01/ip/releases/download/A-Jar/atreides.jar)
- _execute_ 
```
java -jar atreides.jar
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yugan01 committed:
Renaming to Atreides

yugan01 committed:
level 0

yugan01 committed:
Level 1 echo

yugan01 committed:
Add a list

yugan01 committed:
Add Mark done functionality

yugan01 committed:
Add Todos, Events and Deadlines

yugan01 committed:
Run Automated Ui Testing

yugan01 committed:
Change Atreides Logo

yugan01 committed:
Handle some errors

yugan01 committed:
Add Delete functionality and refine some exceptions

yugan01 committed:
Read and write changes to file

yugan01 committed:
Add stashed changes

yugan01 committed:
Add datetime functionality for deadline

Datetime in the format of "MMM dd yyyy HH:mm" or "dd-mm-yyyy"

yugan01 committed:
Add more OOP functionalities

Created more classes for greater abstraction and encapsulation of functions

yugan01 committed:
Divide classes into packages

Commands in command package, tasks in task package and rest in Ui package

yugan01 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

yugan01 committed:
Add Gradle support

yugan01 committed:
Add JUnit test for TaskList and Storage

yugan01 committed:
Create Jar file

created jar executable: atreides.jar

yugan01 committed:
Add Javadoc

yugan01 committed:
Change coding format

yugan01 committed:
Add find functionality to tasks

yugan01 committed:
Merge branch 'branch-A-CodingStandard'

yugan01 committed:
Merge branch 'branch-Level-9'

dasha3412 commented on src/main/java/atreides/command/AddCommand.java:
```diff
@@ -0,0 +1,48 @@
+ */
+
+public class AddCommand implements Command {
+    private final String msg;
```
Should this variable name be written out fully?

dasha3412 commented on src/main/java/atreides/command/AddCommand.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    /**
+     * {@inheritDoc}
```
I like how you inherited the header from the parent class

dasha3412 commented on src/main/java/atreides/command/ErrorCommand.java:
```diff
@@ -0,0 +1,33 @@
+
+    /**
+     * Ui will show the error message
+     * @param tasks
+     * @param ui
+     * @param storage
```
Although these params aren't used, should they still have descriptions for what would usually be the input?

dasha3412 commented on src/main/java/atreides/task/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+
+public class Deadline extends Task {
+
+    protected String by;
```
Should this variable be named more descriptively (such deadline, or timeBy), just to make its purpose clear?

dasha3412 commented on src/main/java/atreides/task/Task.java:
```diff
@@ -0,0 +1,57 @@
+
+    /**
+     * represents whether the task has been completed or not
+     * @param trueString
+     * @param falseString
+     * @return trueString if completed else falseString
```
Should you add a description for these params to avoid confusion for what they represent?

dasha3412 commented on src/main/java/atreides/task/TaskList.java:
```diff
@@ -0,0 +1,148 @@
+        list.add(task);
+    }
+
+    public String find(String description) {
```
Should this method be named findTask(s) or have a header to make it clear what it does?

dasha3412 commented on src/main/java/atreides/ui/Parser.java:
```diff
@@ -0,0 +1,51 @@
+     * @return
+     */
+    public static Command parse(String msg) {
+        Set<String> commands = Set.of("mark", "unmark", "delete", "todo", "event", "deadline", "find");
```
I like how for the commands and the words variable later you used descriptive names, it makes it clear what they are for.

dasha3412 commented on src/main/java/atreides/ui/Response.java:
```diff
@@ -0,0 +1,18 @@
+
+public class Response {
+    String msg;
+    String LINE = "____________________________________________________________";
```
Should this variable be a **static final** variable if it is a constant in screaming snake case (all capitals), or renamed if it is not a constant?

dasha3412 commented on src/main/java/atreides/ui/Ui.java:
```diff
@@ -0,0 +1,47 @@
+ * Represents the logic for the Ui that handles the printing to user and reading user commands
+ */
+public class Ui {
+    private final String LOGO = "          _            _     _\n"
+                             +  "    / \\  | |          (_)   | |\n"
+                             +  "   /   \\ | |_ _ __ ___ _  __| | ___  ___\n"
+                             +  "  / / \\ \\| __| '__/ _ \\ |/ _` |/ _ \\/ __|\n"
+                             +  " / _____ \\ |_| | |  __/ | (_| |  __/\\__ \\\n"
+                             +  "/_/     \\_\\__|_|  \\___|_|\\__,_|\\___||___/\n";
+    private final String INTRO = "Glory to house\n" + LOGO;
+    private final String OUTRO = "Praise be Muad Dib\n" + "GoodBye\n";;
+    private final String ERROR_STRING = "File cannot be found\n";
```
Should these variables all also be ***static***, if they are constant (screaming snake case)?

dasha3412 commented on src/main/java/atreides/ui/Ui.java:
```diff
@@ -0,0 +1,47 @@
+        showError(ERROR_STRING);
+    }
+
+    public void showExit() {
```
I like how consistently you name your methods are verbs and have them in camelCase.

dasha3412 commented on src/test/java/atreides/task/TaskListTest.java:
```diff
@@ -0,0 +1,34 @@
+    }
+
+    @Test
+    public void mark_fail() throws AtreidesException {
```
Should these methods be named in camelCase rather than snake_case?

dasha3412 reviewed (commented):
Looks great 👍 , especially your method naming! Just some small things throughout, mainly with adding more descriptive header comments and including descriptions for params and return variables for those headers.

yugan01 committed:
Rectify checkstyle discrepencies

yugan01 committed:
Merge branch 'branch-A-CheckStyle'

yugan01 committed:
Add GUI for user functionality

yugan01 committed:
Create a new Jar file

yugan01 committed:
Add assertion statements

No assertion statements present to ensure no programming errors

Let's,
* Add assertion statements to ensure no potential bugs in the code

yugan01 committed:
Improve code quality

yugan01 committed:
Add streams

Unnecessary for loop to print out the list of tasks

Let's use a stream to simplify the code to return the string of list of tasks

yugan01 committed:
Merge pull request #1 from yugan01/branch-A-Assertions

Add assertion statements

yugan01 committed:
Merge pull request #2 from yugan01/branch-A-CodeQuality

Improve code quality

yugan01 committed:
Merge pull request #3 from yugan01/branch-A-Streams

Add streams

yugan01 committed:
Inform user of deadline formats

yugan01 committed:
Provide a minimal project template

yugan01 committed:
Update gradle.yml

yugan01 committed:
Add functionality to check for duplicates

yugan01 committed:
Merge branch 'master' of https://github.com/yugan01/ip

yugan01 committed:
Update readme.md and add Ui.png screenshot

yugan01 committed:
Add Javadoc comments

yugan01 committed:
Update Readme and improve GUI

yugan01 committed:
Update ReadMe and function to create text file

yugan01 committed:
Update README.md
